### PR TITLE
ContainerInstance: Adding April Swagger For GA

### PR DESCRIPTION
**Pull request by samkreter**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Review by hovsepm — CHANGES_REQUESTED** [review]
please add a new tag to `specification/containerinstance/resource-manager/readme.md` with the new version

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-04-01/containerInstance.json [hunk 1]
```diff
@@ -0,0 +1,1234 @@
...
+          },
+          "description": "The list of container groups."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URI to fetch the next page of container groups."
+        }
+      }
+    },
+    "Logs": {
```
Comment (hovsepm) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-04-01/containerInstance.json]: >   "Logs": { [](start = 1, length = 12)

This type is used only as return value from the server. It is not supposed to be sent back to server thus all the properties should be marked as read-only.

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-04-01/containerInstance.json [hunk 2]
```diff
@@ -0,0 +1,1234 @@
...
+            "column": {
+              "type": "integer",
+              "description": "The column size of the terminal"
+            }
+          }
+        }
+      }
+    },
+    "ContainerExecResponse": {
+      "description": "The information for the container exec command.",
```
Comment (hovsepm) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-04-01/containerInstance.json]: Same comment here - the type is used only as server response. Please mark all the properties as read-only.

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-04-01/containerInstance.json (hunk 1 above)
Comment (samkreter) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-04-01/containerInstance.json]: Done

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-04-01/containerInstance.json (hunk 2 above)
Comment (samkreter) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-04-01/containerInstance.json]: Done

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-04-01/containerInstance.json [hunk 3]
```diff
@@ -0,0 +1,1234 @@
...
+          "description": "The list of operations."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The URI to fetch the next page of operations."
+        }
+      }
+    },
+    "Operation": {
+      "description": "An operation for Azure Container Instance service.",
```
Comment (hovsepm) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-04-01/containerInstance.json]: same comment - this is used only as server return value. please mark all the properties as read-only.

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-04-01/containerInstance.json [hunk 4]
```diff
@@ -0,0 +1,1234 @@
...
+          "x-ms-enum": {
+            "name": "ContainerInstanceOperationsOrigin",
+            "modelAsString": true
+          }
+        }
+      },
+      "required": [
+          "name",
+          "display"
+      ]
```
Comment (hovsepm) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-04-01/containerInstance.json]: does not make sense to put required fields in the type that is used only as a server response.

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-04-01/containerInstance.json [hunk 5]
```diff
@@ -0,0 +1,1234 @@
...
+          "description": "The volume mounts available to the container instance.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/VolumeMount"
+          }
+        }
+      }
+    },
+    "ContainerState": {
+      "description": "The container instance state.",
```
Comment (hovsepm) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-04-01/containerInstance.json]: the name of this type suggest that it is populated only by server - please mark all the properties as read-only

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-04-01/containerInstance.json [hunk 6]
```diff
@@ -0,0 +1,1234 @@
...
+          "format": "date-time",
+          "description": "The date-time when the container instance state finished."
+        },
+        "detailStatus": {
+          "description": "The human-readable status of the container instance state.",
+          "type": "string"
+        }
+      }
+    },
+    "Event": {
```
Comment (hovsepm) [inline specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-04-01/containerInstance.json]: >Event [](start = 5, length = 5)

same here - user should not modify the events received from server. Please mark properties as read-only.

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-04-01/containerInstance.json (hunk 4 above)
Comment (samkreter) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-04-01/containerInstance.json]: Done

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-04-01/containerInstance.json (hunk 3 above)
Comment (samkreter) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-04-01/containerInstance.json]: Done

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-04-01/containerInstance.json (hunk 6 above)
Comment (samkreter) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-04-01/containerInstance.json]: Done

File: specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-04-01/containerInstance.json (hunk 5 above)
Comment (samkreter) [reply specification/containerinstance/resource-manager/Microsoft.ContainerInstance/stable/2018-04-01/containerInstance.json]: Done

**Comment by sergey-shandar** [comment]
@ravbhatnagar Could you check if it was approved before?

**Comment by hovsepm** [comment]
@samkreter according to swagger spec specifications https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md#fixed-fields-13 `readOnly` attribute can be applied only to `properties` i.e. you should mark each field under `properties` as `readOnly` and not the type itself. Could you revise your pr?

**Comment by samkreter** [comment]
@hovsepm Just finished the changes you requested, but there is a python CI test failing that looks like it is trying to use the Preview version.  Any ideas?

**Review by hovsepm — APPROVED** [review]

**Comment by hovsepm** [comment]
@samkreter once ARM team will sing off on the new stable version we will be able to merge.

**Comment by ravbhatnagar** [comment]
Signing off from ARM side.

**Pull request merged**